framework_crates/bones_ecs/src/entities.rs

```diff
@@ -152,32 +169,56 @@ where
 
 impl<'a, 'q, T: HasSchema> QueryItem for &'a Comp<'q, T> {
     type Iter = ComponentBitsetIterator<'a, T>;
+
     fn apply_bitset(&self, bitset: &mut BitSetVec) {
         bitset.bit_and(self.bitset());
     }
 
+    fn get_single_with_bitset(
+        self,
+        _bitset: Rc<BitSetVec>,
+    ) -> Result<<Self::Iter as Iterator>::Item, QuerySingleError> {
+        ComponentStore::get_single(&**self)
```

Comment:
I was a bit confused why this code does not use the bitset here, but I see how this works, think it does the correct thing, but wondering if we can make this work with bitset applied here.

Perhaps use a `ComponentStore::get_single_with_bitset` that checks if there is one item in component store (and only one) that also is included in bitset.

`ComponentStore::get_single` iterates over the whole component store which makes sense in isolation, but wondering if a new `ComponentStore::get_single_with_bitset` could use bitset iterator to skip checking items not included in bitset much like you did for `get_single_with_bitset`for the multi-component query item.

By passing in bitset, it should allow  `entities.get_single_with` code to only use `get_single_with_bitset` on query item instead of checking entity bitset first, even for this query item that is only single component.

Reply:
Great catch! I would actually consider this a bug. It seems like you get the gist of the behavior here but I'm going to explain the bug in more detail here for reference.

An `Entity` *can be created manually* so it would be possible to e.g. have only 1 entity with a component `A`, but do a `QueryItem::get_single_with_bitset` for `A` with a bit enabled on the bitset set for an entity that doesn't exist. It would return the component for the real entity when it should return `Err(QuerySingleError::NoEntities)`.

I've added a regression test for this.

Reply:
Oh yeah I missed that part of it! Sounds good thank you!